fastify.js

```diff
@@ -26,38 +26,24 @@ function build () {
   }
 
   function get (url, schema, handler) {
-    if (!handler && typeof schema === 'function') {
-      handler = schema
-      schema = {}
-    }
-    return route({
-      method: 'GET',
-      url: url,
-      schema: schema,
-      handler: handler
-    })
+    return _route('GET', url, schema, handler)
   }
 
   function post (url, schema, handler) {
-    if (!handler && typeof schema === 'function') {
-      handler = schema
-      schema = {}
-    }
-    return route({
-      method: 'POST',
-      url: url,
-      schema: schema,
-      handler: handler
-    })
+    return _route('POST', url, schema, handler)
   }
 
   function put (url, schema, handler) {
+    return _route('PUT', url, schema, handler)
+  }
+
+  function _route (method, url, schema, handler) {
     if (!handler && typeof schema === 'function') {
       handler = schema
       schema = {}
     }
     return route({
-      method: 'PUT',
+      method: method,
```

Comment:
I've run the tests under node v4 and only throw.test.js failed (because of the destructuring assignment), with a super tiny fix we can make the code base v4 compatible.
Should we support v4? In my opinion, yes.